### PR TITLE
docs: update package versions, add ROCm, edit to style guide

### PR DESCRIPTION
**Pull request by gh-determined-ai**
## Checklist

- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
See [Release Note](../docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

[DET-1234]: https://determinedai.atlassian.net/browse/DET-1234?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: docs/prepare-environment/custom-env.txt [hunk 1]
```diff
@@ -8,22 +8,22 @@ Determined launches workloads using Docker containers. By default, workloads exe
...
 ***********************
  Environment Variables
 ***********************
 
-For both trial runners and commands, Determined allows users to configure the environment variables
-inside the container through the ``environment.environment_variables`` configuration field of the
-:ref:`experiment <experiment-configuration>` or :ref:`task <command-notebook-configuration>` config.
-The format is a list of strings in the format ``NAME=VALUE``:
+For both trial runners and commands, you configure the environment variables inside the container
+using the :ref:`experiment <experiment-configuration>` or :ref:`task
```
Comment (ioga) [inline docs/prepare-environment/custom-env.txt:24]: I really wish we had anchors for the individual config options.

File: docs/prepare-environment/custom-env.txt [hunk 2]
```diff
@@ -59,96 +59,97 @@ It is also possible to set these variables for each accelerator type separately:
...
-   ``determinedai/environments:py-3.8-pytorch-1.10-lightning-1.5-tf-2.8-cpu-0.17.15`` for GPU and
-   CPU respectively.
++-------------+---------------------------------------------------------------------------------------+
+| Environment | File Name                                                                             |
++=============+=======================================================================================+
+| CPUs        | ``determinedai/environments:py-3.8-pytorch-1.10-lightning-1.5-tf-2.8-cpu-0.17.12``    |
++-------------+---------------------------------------------------------------------------------------+
+| Nvidia GPUs | ``determinedai/environments:cuda-11.3-pytorch-1.10-lightning-1.5-tf-2.8-gpu-0.17.12`` |
++-------------+---------------------------------------------------------------------------------------+
+| ROCm        | ``determinedai/environments:rocm-4.2-pytorch-1.9-tf-2.5-rocm-0.17.12``                |
```
Comment (ioga) [inline docs/prepare-environment/custom-env.txt:108]: using either CUDA & ROCm or Nvidia GPUs & AMD GPUs dichotomy would probably be more correct

File: docs/prepare-environment/custom-env.txt [hunk 3]
```diff
@@ -59,96 +59,97 @@ It is also possible to set these variables for each accelerator type separately:
...
 -  ``determinedai/environments:cuda-10.1-pytorch-1.4-tf-2.2-gpu-067db2b``
 
+The performance of some models can vary with different CUDA versions.
+
+Container Contents
+==================
+
+Experiments and tasks are run in containers that include the following packages:
+
+-  CUDA 11.1
```
Comment (ioga) [inline docs/prepare-environment/custom-env.txt:126]: default CUDA image currently has 11.3.

To be fair, this whole paragraph may not be super helpful: according to the previous section, tasks may run in CPU/CUDA/ROCm containers which would not have CUDA at all, or have ROCm.

**Review by ioga — APPROVED** [review]

File: docs/prepare-environment/custom-env.txt (hunk 1 above)
Comment (gh-determined-ai) [reply docs/prepare-environment/custom-env.txt:24]: As long as the anchors are headings, that's fine. I'll make them headings if they're not and if it makes sense.

File: docs/prepare-environment/custom-env.txt [hunk 4]
```diff
@@ -59,96 +59,97 @@ It is also possible to set these variables for each accelerator type separately:
...
 
-If a file named ``startup-hook.sh`` exists at the top level of your model definition directory,
-Determined will automatically execute this file during the startup of every Docker container. This
-occurs before any Python interpreters are launched or any deep learning operations are performed;
-this allows the startup hook to customize the container environment, install additional
-dependencies, download data sets, or do practically anything else that you can do in a shell script.
+If a ``startup-hook.sh`` file exists in the top level of your model definition directory, this file
+is automatically run with every Docker container startup. This occurs before any Python interpreters
+are launched or deep learning operations are performed, permitting the startup hook to customize the
+container environment, install additional dependencies, download data sets, or do almost anything
```
[the remark this answers is not in the document]
Comment (gh-determined-ai) [reply docs/prepare-environment/custom-env.txt:65]: Yeah, I'll make it less folksy.

File: docs/prepare-environment/custom-env.txt (hunk 2 above)
Comment (gh-determined-ai) [reply docs/prepare-environment/custom-env.txt:108]: @shiyuann indicated these versions were the same as listed in the Experiment Configuration _image_ field (likely an oversight). I'll make it 0.17.15 in both locations. Eventually, there should be only one reference, not duplicated.

File: docs/prepare-environment/custom-env.txt [hunk 5]
```diff
@@ -59,96 +59,97 @@ It is also possible to set these variables for each accelerator type separately:
...
+| ROCm        | ``determinedai/environments:rocm-4.2-pytorch-1.9-tf-2.5-rocm-0.17.12``                |
++-------------+---------------------------------------------------------------------------------------+
 
-Older Images
-============
-
-Images that provide older versions of the frameworks are still available and supported. Note that
-the performance of some models can vary with different CUDA versions.
+Supported legacy Images
+=======================
```
[the remark this answers is not in the document]
Comment (gh-determined-ai) [reply docs/prepare-environment/custom-env.txt:112]: Will do.

File: docs/prepare-environment/custom-env.txt (hunk 1 above)
Comment (gh-determined-ai) [reply docs/prepare-environment/custom-env.txt:24]: I'll add that to the wish list. But, it's generally an unmaintainable feature, which is why you only link to headings and even that's problematic. Overall, there's too much linking in the Determined docs but that first needs to be addressed at a higher level by restructuring, then by defining a sane linking model.

File: docs/prepare-environment/custom-env.txt (hunk 4 above)
[the remark this answers is not in the document]
Comment (gh-determined-ai) [reply docs/prepare-environment/custom-env.txt:65]: Yeah, will reword the folksy wording.

File: docs/prepare-environment/custom-env.txt (hunk 2 above)
[the remark this answers is not in the document]
Comment (gh-determined-ai) [reply docs/prepare-environment/custom-env.txt:108]: Will also update the Environment Configuration image field. Eventually, there should be only one source of truth, not duplicated.

[the remark this answers is not in the document]
Comment (gh-determined-ai) [reply docs/prepare-environment/custom-env.txt:108]: I see it's already updated in the Environment Configuration but not yet published. A self-revealing problem.

Comment (gh-determined-ai) [reply docs/prepare-environment/custom-env.txt:108]: @ioga  Not sure what you mean here. These 3 were taken from the Experiment Configuration _image_ documentation.

File: docs/prepare-environment/custom-env.txt (hunk 3 above)
Comment (gh-determined-ai) [reply docs/prepare-environment/custom-env.txt:126]: Will delete. Another unmaintainable item. If significant, these should be listed in a clear reference doc and not buried here.

File: docs/prepare-environment/custom-env.txt (hunk 2 above)
Comment (gh-determined-ai) [reply docs/prepare-environment/custom-env.txt:108]: @ioga Please, disregard. I see what you mean.

**Pull request merged**